src/main/java/com/iexec/common/chain/IexecHubAbstractService.java

```diff
@@ -111,11 +110,43 @@ public IexecHubAbstractService(Credentials credentials,
                                    int blockTime,
                                    int nbBlocksToWaitPerRetry,
                                    int maxRetries) {
+        this(
+                credentials,
+                web3jAbstractService,
+                iexecHubAddress,
+                Duration.ofMillis(blockTime > 0 ? blockTime : DEFAULT_BLOCK_TIME),
+                nbBlocksToWaitPerRetry,
+                maxRetries
+        );
+    }
+
+    /**
+     * Base constructor for the IexecHubAbstractService
+     * @param credentials credentials for sending transaction
+     * @param web3jAbstractService custom web3j service
+     * @param iexecHubAddress address of the iExec Hub contract
+     * @param blockTime block time as a duration
+     * @param nbBlocksToWaitPerRetry nb block to wait per retry
+     * @param maxRetries maximum reties
+     */
+    public IexecHubAbstractService(Credentials credentials,
+                                   Web3jAbstractService web3jAbstractService,
+                                   String iexecHubAddress,
+                                   Duration blockTime,
+                                   int nbBlocksToWaitPerRetry,
+                                   int maxRetries) {
         this.credentials = credentials;
         this.web3jAbstractService = web3jAbstractService;
         this.iexecHubAddress = iexecHubAddress;
         this.nbBlocksToWaitPerRetry = nbBlocksToWaitPerRetry;
-        this.retryDelay = nbBlocksToWaitPerRetry * blockTime;
+        if (blockTime == null || blockTime.toMillis() <= 0) {
+            log.warn("Blocktime value is incorrect, using default value [blockTime:{}, DEFAULT_BLOCK_TIME:{}]",
+                    blockTime == null ? "null" : blockTime, DEFAULT_BLOCK_TIME);
```

Comment:
There is no need for this
```suggestion
                    blockTime, DEFAULT_BLOCK_TIME);
```

Reply:
You're right! Updated, thanks :)